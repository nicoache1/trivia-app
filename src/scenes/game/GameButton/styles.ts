import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: 'white',
    borderColor: 'black',
    borderRadius: 40,
    elevation: 11,
    height: 110,
    justifyContent: 'center',
    shadowColor: '#000',
    shadowOffset: {
      height: 5,
      width: 0,
    },
    shadowOpacity: 0.36,
    shadowRadius: 6.68,
    width: 110,
  },
  buttonImage: {
    height: 80,
    width: 80,
  },
})
