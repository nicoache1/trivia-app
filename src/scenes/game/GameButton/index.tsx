import React from 'react'
import { Image, ImageSourcePropType, Pressable } from 'react-native'

import { styles } from './styles'

interface Props {
  onPress: () => void
  imageSource: ImageSourcePropType
}

export const GameButton: React.FC<Props> = ({ onPress, imageSource }) => {
  return (
    <Pressable
      testID="gameButton"
      onPress={onPress}
      style={({ pressed }) => [
        {
          opacity: pressed ? 0.7 : 1,
        },
        styles.button,
      ]}>
      <Image source={imageSource} style={styles.buttonImage} />
    </Pressable>
  )
}
