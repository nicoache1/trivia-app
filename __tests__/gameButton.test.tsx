import { fireEvent, render, screen } from '@testing-library/react-native'
import Image from 'assets/check.png'
import React from 'react'

import { GameButton } from '../src/scenes/game/GameButton/index'

jest.setTimeout(7000)

const image = Image
const mockFn = jest.fn()

test('button gets pressed', () => {
  render(<GameButton onPress={mockFn} imageSource={image} />)

  fireEvent.press(screen.getByTestId('gameButton'))

  expect(mockFn).toBeCalledTimes(1)
}, 10000)

test('Expect GameButton to be rendered correctly', async () => {
  const renderedSnapshot = render(
    <GameButton onPress={mockFn} imageSource={image} />,
  ).toJSON()

  expect(renderedSnapshot).toMatchSnapshot()
})
