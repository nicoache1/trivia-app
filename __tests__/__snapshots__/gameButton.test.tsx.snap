// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect GameButton to be rendered correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "opacity": 1,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 40,
        "elevation": 11,
        "height": 110,
        "justifyContent": "center",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.36,
        "shadowRadius": 6.68,
        "width": 110,
      },
    ]
  }
  testID="gameButton"
>
  <Image
    source={1}
    style={
      Object {
        "height": 80,
        "width": 80,
      }
    }
  />
</View>
`;
